.circleci/config.yml

```diff
@@ -339,6 +339,34 @@ jobs:
             RUST_BACKTRACE=1 $CARGO $CARGOFLAGS build -j 16 -p transaction-builder -p move-vm-types --target powerpc-unknown-linux-gnu
       - save_sccache
       - build_teardown
+  build-individual-crates:
+    executor: build-executor
+    parallelism: 7
+    description: Build all top level individual crates
+    steps:
+      - build_setup
+      - restore_cargo_package_cache
+      - install_sccache
+      - restore_sccache
+      - run:
+          name: Determine crate targets for this container.
+          command: |
+            for x in $(find . -name Cargo.toml); do dirname $x >> /tmp/crates; done
+            echo "Found $(wc -l /tmp/crates) crates"
+            cat /tmp/crates | circleci tests split > /tmp/crates_to_build
+            echo -e "This container will build these crates\n$(cat /tmp/crates_to_build)"
+      - run:
+          name: Build assigned crates in this container.
+          command: |
+            set -x
+            _pwd=$PWD
+            for crate in $(cat /tmp/crates_to_build) ; do
+              cd $_pwd/$crate
+              RUST_BACKTRACE=1 $CARGO $CARGOFLAGS build
+              RUST_BACKTRACE=1 $CARGO $CARGOFLAGS test --no-run
```

Comment:
looks like xtest supports no run, if you use it here it's a good entry point for global behavior change on tests.

Reply:
Will update to xtest.